image_registry.yaml

```diff
@@ -1,342 +1,817 @@
+    - 1.26.0-distroless
+    - 1.25.3-distroless
+    - 1.25.2-distroless
+    - 1.25.1-distroless
+    - 1.25.0-distroless
+    - 1.24.6-distroless
     - 1.24.5-distroless
+    - 1.23.3-distroless
+    - 1.23.2-distroless
+    - 1.23.1-distroless
+    - 1.22.3-distroless
+  tag_regex: ^\d+\.\d+\.\d+-distroless$
 jupyter_datascience_lab:
-  default_tag: lab-4.4.0
   name: quay.io/jupyter/datascience-notebook
+  default_tag: lab-4.4.3
   tags:
+    - lab-4.4.3
+    - lab-4.4.2
+    - lab-4.4.1
     - lab-4.4.0
-  tag_regex: '^lab-\d+\.\d+\.\d+$'
+  tag_regex: ^lab-\d+\.\d+\.\d+$
 karpenter:
-  default_tag: 1.2.0
   name: public.ecr.aws/karpenter/controller
-  tags:
-    - 0.37.0
-    - 1.0.8
+  default_tag: 1.5.0
+  tags:
+    - 1.5.0
+    - 1.4.0
+    - 1.3.4
+    - 1.3.3
+    - 1.3.2
+    - 1.3.1
+    - 1.3.0
+    - 1.2.4
+    - 1.2.3
+    - 1.2.2
+    - 1.2.1
     - 1.2.0
-  tag_regex: '^\d+\.\d+\.\d+$'
+    - 1.0.8
+    - 0.37.0
+  tag_regex: ^\d+\.\d+\.\d+$
 keycloak:
-  default_tag: 26.2.0
   name: quay.io/keycloak/keycloak
+  default_tag: 26.2.5
   tags:
-    - 25.0.2
-    - 25.0.6
+    - 26.2.5
+    - 26.2.4
+    - 26.2.3
+    - 26.2.2
+    - 26.2.1
     - 26.2.0
-  tag_regex: '^\d+\.\d+\.\d+$'
-# Be sure to check:
-# https://kiali.io/docs/installation/installation-guide/prerequisites/#version-compatibility
+    - 25.0.6
+    - 25.0.2
+  tag_regex: ^\d+\.\d+\.\d+$
 kiali:
-  default_tag: v2.8.0
   name: quay.io/kiali/kiali
+  default_tag: v1.89.7
   tags:
-    - v1.87.0
     - v1.89.7
+    - v1.87.0
     - v2.8.0
-  tag_regex: '^v\d+\.\d+\.\d+$'
+  tag_regex: ^v\d+\.\d+\.\d+$
 kiwigrid_sidecar:
-  default_tag: 1.30.3
   name: quay.io/kiwigrid/k8s-sidecar
+  default_tag: 1.30.3
   tags:
-    - 1.27.4
     - 1.30.3
-  tag_regex: '^\d+\.\d+\.\d+$'
+    - 1.27.4
+  tag_regex: ^\d+\.\d+\.\d+$
 knative_istio_controller:
-  default_tag: &knative_istio_default_tag v1.17.0
   name: gcr.io/knative-releases/knative.dev/net-istio/cmd/controller
-  tag_regex: '^v\d+\.\d+\.\d+$'
-  tags: &knative_istio_allowed_tags
-    - v1.15.1
+  default_tag: v0.26.0
+  tags:
+    - v0.26.0
+    - v0.25.1
+    - v0.25.0
+    - v0.24.1
+    - v0.24.0
+    - v0.23.2
+    - v0.23.1
+    - v0.23.0
+    - v0.22.2
+    - v0.22.1
+    - v0.22.0
+    - v0.21.0
+    - v0.20.0
+    - v0.19.0
+    - v0.18.1
+    - v0.18.0
+    - v1.18.0
+    - v0.17.1
+    - v0.17.0
     - v1.17.0
+    - v1.15.1
+  tag_regex: ^v\d+\.\d+\.\d+$
 knative_istio_webhook:
-  default_tag: *knative_istio_default_tag
   name: gcr.io/knative-releases/knative.dev/net-istio/cmd/webhook
-  tag_regex: '^v\d+\.\d+\.\d+$'
-  tags: *knative_istio_allowed_tags
+  default_tag: v0.26.0
+  tags:
+    - v0.26.0
+    - v0.25.1
+    - v0.25.0
+    - v0.24.1
+    - v0.24.0
+    - v0.23.2
+    - v0.23.1
+    - v0.23.0
+    - v0.22.2
+    - v0.22.1
+    - v0.22.0
+    - v0.21.0
+    - v0.20.0
+    - v0.19.0
+    - v0.18.1
+    - v0.18.0
+    - v1.18.0
+    - v0.17.1
+    - v0.17.0
+    - v1.17.0
+    - v1.15.1
```

Comment:
Same here. v1 before v0.

Reply:
Letters before the numbers was messing up my parsing. I added some tests.